Source/JavaScriptCore/dfg/DFGFixupPhase.cpp

```diff
@@ -3947,14 +3947,6 @@ class FixupPhase : public Phase {
             return;
         }
 
-        if (node->child1()->shouldSpeculateStringOrOther()) {
```

Comment:
We should add a different node or something. It is particularly important, not only because it is just using ToString, but also it makes effect model simpler so that we can encourage the other optimizations around this node.